src/GraphQL.Relay.Test/Utilities/RelayPaginationTests.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using GraphQL.Relay.Types;
+using GraphQL.Relay.Utilities;
+using Xunit;
+
+namespace GraphQL.Relay.Test.Utilities
+{
+    public class RelayPaginationTests
+    {
+        [Fact]
+        public void Apply_IfNoPaginationArgumentsAreProvided_ReturnsFullRange()
+        {
+            var range = RelayPagination.CalculateEdgeRange(10);
+            Assert.Equal(0, range.StartOffset);
+            Assert.Equal(9, range.EndOffset);
+        }
+
+
+        [Fact]
+        public void Apply_IfAfterIsProvided_ReturnsAllEdgesAfterArgument()
+        {
+            var range = RelayPagination.CalculateEdgeRange(10, after: ConnectionUtils.OffsetToCursor(5));
+            Assert.Equal(6, range.StartOffset);
+            Assert.Equal(9, range.EndOffset);
+        }
+
+
+        [Fact]
+        public void Apply_IfBeforeIsProvided_ReturnsAllEdgesBeforeArgument()
+        {
+            var range = RelayPagination.CalculateEdgeRange(10, before: ConnectionUtils.OffsetToCursor(5));
+            Assert.Equal(0, range.StartOffset);
+            Assert.Equal(4, range.EndOffset);
+        }
+
+        [Theory]
+        [InlineData(20, -3)]
+        [InlineData(20, -2)]
+        [InlineData(20, 21)]
+        [InlineData(20, 22)]
+        public void Apply_IfAfterIsOutOfRange_SilentlyUses0AsBoundary(int edgeCount, int afterOffset)
+        {
+            var range = RelayPagination.CalculateEdgeRange(edgeCount, after: ConnectionUtils.OffsetToCursor(afterOffset));
+            Assert.Equal(0, range.StartOffset);
+        }
+
+        [Theory]
+        [InlineData(20, -3)]
+        [InlineData(20, -2)]
+        [InlineData(20, 21)]
+        [InlineData(20, 22)]
+        public void Apply_IfBeforeIsOutOfRange_SilentlyUsesTotalCountAsBoundary(int edgeCount, int afterOffset)
```

Comment:
might be worth adding some cases for after at the end or start of a range, or length=1 range (unless i've missed those tests)

Reply:
I've added a couple more test cases to check for more edge cases (and I indeed found a minor bug in the EdgeRange constructor)